src/iotEdgeExplorer.ts

```diff
@@ -116,6 +116,36 @@ export class IoTEdgeExplorer extends BaseExplorer {
         return true;
     }
 
+    private isValidCreateOptions(desiredProperties: any): boolean {
+        return this.isValidCreateOptionsHepler(desiredProperties.systemModules) && this.isValidCreateOptionsHepler(desiredProperties.modules);
+    }
+
+    private isValidCreateOptionsHepler(modules: any): boolean {
+        for (const moduleName in modules) {
+            if (modules.hasOwnProperty(moduleName)) {
+                try {
+                    const createOptions = modules[moduleName].settings.createOptions;
+                    if (createOptions) {
+                        this.checkJsonString(createOptions);
+                    }
+                } catch (error) {
+                    vscode.window.showErrorMessage(`Cannot convert createOptions of module "${moduleName}" as json object: ${error.message}`);

```

Comment:
How about change to "createOptions of <module> is not a valid JSON string: ${error.message}"

Reply:
OK

Reply:
Updated